docs/functions-dev.md

```diff
@@ -42,3 +42,66 @@ To do so set the `--inspect` Node.js option when starting the functions server:
 - On Mac/Linux run `NODE_OPTIONS=--inspect netlify functions:serve`
 
 Then attach any Node.js debugger to the CLI process to debug your functions.
+
+
+## VS Code Debugging
+
+Put the below configuration in your repo's `.vscode/launch.json`.
+
+Then in `VS Code`:
+1. Open `Run and Debug` Sidebar. Make sure "Caught Exceptions" is deactivated.
+2. In the dropdown, select `Fullstack Debugging`
+3. Run the debugger. A chrome window should open. Wait until content is loaded (may take a few seconds for server to be ready).
+
+```jsonc
+{
+  // Use IntelliSense to learn about possible attributes.
+  // Hover to view descriptions of existing attributes.
+  // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
+  "version": "0.2.0",
+  // ---- FULLSTACK NETLIFY DEV DEBUGGING ----
+  // 1. Make sure "Caught Exceptions" is deactivated in your VS Code debugger
+  // 2. Run 'Fullstack Debugging' from the debug interface
+  // 3. A chrome browser window will open, however it may take a few seconds
+  // before the server is ready and content is displayed.
+  "compounds": [
```

Comment:
Do we need the Chrome client? This ties the configuration to a specific browser which might not be installed on some machines. Also I was prompted to install a VS Code extension to get it working.

I was able to debug the CLI by using the following minimal config:
```json
{
  "version": "0.2.0",
  "configurations": [
    {
      "type": "node",
      "request": "launch",
      "name": "Launch Program",
      "skipFiles": ["<node_internals>/**"],
      "program": "${workspaceFolder}/node_modules/.bin/netlify",
      "args": ["dev"]
    }
  ]
}
```

Also we should instruct users to install the CLI locally via `npm install --save-dev netlify-cli` for this to work

Reply:
This section is about functions debugging and I see it serving 2 purposes:
1. Debugging only functions with `netlify functions:serve`
2. Debugging functions with redirects, headers, etc with `netlify dev`

As for frontend debugging, I don't think it's related to the original issue https://github.com/netlify/cli/issues/409, so can we descope it from this PR?

I agree this is a valid use case, so having another section for it would makes sense.
In that section I prefer to link to https://code.visualstudio.com/blogs/2016/02/23/introducing-chrome-debugger-for-vs-code or https://code.visualstudio.com/docs/editor/debugging instead of duplicating content that might go stale.

Please let me know if that makes sense, and sorry for not being clear enough on this from the start.

Reply:
Thanks for clarifying that as indeed I overlooked that the page is called "Serving functions locally" with only the subsection being about local function debugging.

Since you also agree that having fullstack debugging for netlify-cli is valid, I would then rather propose to simply spawn out the whole debugging topic into it's own page where we can have both functions (backend) and frontend (chrome client) debugging mentioned. What do you think ?

Reply:
> Since you also agree that having fullstack debugging for netlify-cli is valid, I would then rather propose to simply spawn out the whole debugging topic into it's own page where we can have both functions (backend) and frontend (chrome client) debugging mentioned. What do you think

Yes that makes sense.